pkgs/by-name/xa/xar/package.nix

```diff
@@ -0,0 +1,175 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  applyPatches,
+
+  autoreconfHook,
+
+  # Required dependencies.
+  zlib,
+  libxml2,
+
+  # Optional dependencies.
+  withExt2 ? lib.meta.availableOn stdenv.hostPlatform e2fsprogs,
+  e2fsprogs,
+  withAcl ? lib.meta.availableOn stdenv.hostPlatform acl,
+  acl,
+  withBzip2 ? true,
+  bzip2,
+  withLzma ? true,
+  xz, # lzma
+
+  # Platform-specific dependencies.
+  musl-fts,
+  openssl,
+  darwin, # CommonCrypto
+
+  gitUpdater,
+
+  # for tests
+  xar,
+  testers,
+  python3,
+  libxslt, # xsltproc
+  runCommand,
+  makeWrapper,
+  buildPackages,
+}:
+let
+  pname = "xar";
+  version = "498";
+
+  pythonForTests = py: py.withPackages (p: [ p.xattr ]);
+
+  src = applyPatches {
+    name = "source";
+    src = fetchurl {
+      url = "https://opensource.apple.com/tarballs/xar/xar-${version}.tar.gz";
+      hash = "sha256-nO5PgLls9ZLMxUWk/dUeTaSlvTtHNJAWN9Z7BD7/PHU=";
+    };
+    patches = lib.filesystem.listFilesRecursive ./patches;
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version src;
+
+  sourceRoot = "source/xar";
+
+  outputs = [
+    "out"
+    "lib"
+    "dev"
+  ];
+
+  strictDeps = true;
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  # For some reason libxml2 package headers are in subdirectory and thus aren’t
+  # picked up by stdenv’s C compiler wrapper (see ccWrapper_addCVars). This
+  # doesn’t really belong here and either should be part of libxml2 package or
+  # libxml2 in Nixpkgs can just fix their header paths.
+  env.NIX_CFLAGS_COMPILE = "-isystem ${libxml2.dev}/include/libxml2";
```

Comment:
Yes, but I believe that in this case it’s actually an issue with Nixpkgs package because it’s namespaced twice, i.e. `include/libxml2/libxml/xmlwriter.h` instead of `include/libxml/xmlwriter.h`…

The libxml’s project documentation uses `#include <libxml/xmlwriter.h>` in examples as well and that just doesn’t work with C compiler wrapper and hooks in Nixpkgs because of that unnecessary namespacing (because stdenv only adds top-level `include` directory for each package to the C header search path).

Reply:
Uh, yeah, that does indeed sound messed up and we should fix it. Sorry for assuming something more normal was going on here.

I guess this happens to work because most things use `pkg-config` to find the headers?

Reply:
>I guess this happens to work because most things use `pkg-config` to find the headers?

Yes, though some other packages set `NIX_CFLAGS_COMPILE` as well. E.g. https://github.com/NixOS/nixpkgs/blob/2e11a22606820ece544b1b7f14de9ae32392ef9a/pkgs/servers/sql/postgresql/generic.nix#L90-L92

Uh, yes, libxml2 also has its own `xml2-config` program that is similar to pkgconf/pkg-config and usually does not work when cross-compiling…

I’m not really sure if there is some knob in libxml2 build system to put the headers in `$dev/include` instead of `$dev/include/libxml2`, but at the very least it should be possible to just move the headers post install and add a symlink for compatibility.
```
mv $dev/include/libxml2 libxml2
rmdir $dev/include
mv libxml2 $dev/include
ln -s . $dev/include/libxml2
```